api/src/test/java/org/openmrs/api/impl/ObsServiceImplTest.java

```diff
@@ -0,0 +1,77 @@
+package org.openmrs.api.impl;
+
+import org.junit.Test;
+import org.openmrs.Concept;
+import org.openmrs.ConceptName;
+import org.openmrs.Encounter;
+import org.openmrs.Location;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.Person;
+import org.openmrs.aop.RequiredDataAdvice;
+import org.openmrs.api.*;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.db.ObsDAO;
+import org.openmrs.api.handler.SaveHandler;
+import org.openmrs.obs.ComplexData;
+import org.openmrs.obs.ComplexObsHandler;
+import org.openmrs.obs.handler.AbstractHandler;
+import org.openmrs.test.Verifies;
+import org.openmrs.util.OpenmrsClassLoader;
+import org.openmrs.util.OpenmrsConstants.PERSON_TYPE;
+import org.openmrs.util.OpenmrsUtil;
+import org.openmrs.util.PrivilegeConstants;
+import org.springframework.transaction.annotation.Transactional;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import java.io.File;
+import org.openmrs.Cohort;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Vector;
+import org.openmrs.api.impl.ObsServiceImpl;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.Test;
+import org.openmrs.Cohort;
+import org.openmrs.api.context.Context;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.openmrs.util.OpenmrsUtil;
+
+
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class ObsServiceImplTest extends ObsServiceTest {
+
+    @Test
+    @Verifies(value = "should add person with encounter patient for a new obs", method = "saveObs(Obs,String)")
+    public void saveObs_shouldUpdateObsPersonValueWithEncounterPatientForNewObs() throws Exception {
```

Comment:
Why do you think this does not belong here https://github.com/openmrs/openmrs-core/blob/master/api/src/test/java/org/openmrs/api/ObsServiceTest.java